plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2186,6 +2185,16 @@ So if getMinSpeed() returns null we fall back to getSpeed().
 
         vm.addComp(devices);
 
+        // Add extra configuration to User VM Domain XML before starting
+        if (vmTO.getType() != VirtualMachine.Type.User) {
+            final GenericXML genericXML = new GenericXML();
+            String extraConfig = vmTO.getExtraConfig();
+            if (extraConfig != null) {
+                genericXML.setExtraConfig(extraConfig);
+                vm.addComp(genericXML);
```

Comment:
Nice. Have you tested adding the component without the genericXML wrapper? I think I saw it supported strings

Reply:
Correct, I tried it and it works just as the same.